torch/quantization/fx/_equalize.py

```diff
@@ -0,0 +1,173 @@
+        # TODO: Allow for convoluational layers
+        if not (x_orig.ndim == 2 and w_orig.ndim == 2 and x_orig.shape[1] == w_orig.shape[1]):
+            raise ValueError(
+                "Input and Weight must have the same column dimension. " +
+                f"Found {x_orig.shape} and {w_orig.shape} instead."
+            )
+
+        return self._forward(x_orig, w_orig)
+
+    def _forward(self, x_orig, w_orig):
+        r"""
+        Calculates the min/max values of each input column, weight column, and weight row.
+        """
+
+        x_orig = self.input_obs(x_orig)
+        w_orig = self.weight_col_obs(w_orig)
+        w_orig = self.weight_row_obs(w_orig)
+
+        # Calculate the column indices of the min/max weight in each row
+        num_row, _ = w_orig.shape
+        min_weights_ind = []
+        max_weights_ind = []
+        for i in range(num_row):
+            min_weights_ind.append(torch.where(w_orig[i] == self.weight_row_obs.min_vals[i])[0][0])
+            max_weights_ind.append(torch.where(w_orig[i] == self.weight_row_obs.max_vals[i])[0][0])
+        self.min_weights_ind = torch.tensor(min_weights_ind)
+        self.max_weights_ind = torch.tensor(max_weights_ind)
+
+        return x_orig, w_orig
+
+    def get_input_minmax(self):
+        return (self.input_obs.min_vals, self.input_obs.max_vals)
+
+    def get_weight_col_minmax(self):
+        return (self.weight_col_obs.min_vals, self.weight_col_obs.max_vals)
+
+    def get_weight_row_minmax(self):
+        return (self.weight_row_obs.min_vals, self.weight_row_obs.max_vals)
+
+    def calculate_equalization_scale(self) -> torch.Tensor:
+        (min_inputs, max_inputs) = self.get_input_minmax()
+        (min_weights, max_weights) = self.get_weight_col_minmax()
+
+        self.equalization_scale = torch.sqrt((max_weights - min_weights) / (max_inputs - min_inputs))
+        return self.equalization_scale
+
+    def calculate_qparams(self):
+        r"""
+        Returns the scale/zero_point for the input and weight rows
+        """
+
+        if self.equalization_scale.nelement() == 0:
+            warnings.warn(
+                "Must call calculate_scale before calling calculate_qparams.\
+                Returning default scale and zero point. "
+            )
+            return torch.tensor([1.0]), torch.tensor([0]), torch.tensor([1.0]), torch.tensor([0])
+
+        if self.min_weights_ind is None or self.max_weights_ind is None:
+            warnings.warn(
+                "Must find the column indicies of the minimum of each row in the \
+                weights in order to calculate the qparams calculate the \
+                qparams. Returning default scale and zero point. "
+            )
+            return torch.tensor([1.0]), torch.tensor([0]), torch.tensor([1.0]), torch.tensor([0])
+
+        (min_inputs, max_inputs) = self.get_input_minmax()
+        min_input_ind = torch.argmin(min_inputs)
+        max_input_ind = torch.argmax(max_inputs)
```

Comment:
let's multiply the equalization_scale with the min_inputs/max_inputs and then compute the new min/max for the inputs.
So we can do torch.argmin(min_inputs * eq_scale) instead of torch.argmin(min_inputs) * eq_scale[idx]